src/snowflake/cli/api/project/schemas/entities/common.py

```diff
@@ -59,6 +61,27 @@ def get_type(cls) -> str:
         return cls.model_fields["type"].annotation.__args__[0]
 
     meta: Optional[MetaField] = Field(title="Meta fields", default=None)
+    identifier: Optional[Union[Identifier | str]] = Field(
+        title="Entity identifier", default=None
+    )
+    _key: Optional[str] = PrivateAttr(default=None)
+
+    def set_key(self, value: str):
+        self._key = value
+
+    def validate_identifier(self):
+        """Helper that's used by ProjectDefinition validator."""
+        if not self._key and not self.identifier:
```

Comment:
will this ever be true in practice given that we always set the key first?

Reply:
When you create entitiy outside context it may be true